### ISSUE TITLE
Redundant double debug registrations for plugins

### ISSUE THREAD

**Issue by colin-grant-work** [eclipse-theia/theia#10835]
### Bug Description:

We register debuggers once [in the contribution handler](https://github.com/eclipse-theia/theia/blob/41a2f5b8d330944498a8e7715d8a30921a73f5d6/packages/plugin-ext/src/main/browser/plugin-contribution-handler.ts#L331-L339) and once [in the plugin context](https://github.com/eclipse-theia/theia/blob/41a2f5b8d330944498a8e7715d8a30921a73f5d6/packages/plugin-ext/src/plugin/plugin-context.ts#L802-L804).

In the `PluginDebugService`, the first registration puts the contribution [in the `debuggers` array](https://github.com/eclipse-theia/theia/blob/41a2f5b8d330944498a8e7715d8a30921a73f5d6/packages/plugin-ext/src/main/browser/debug/plugin-debug-service.ts#L181-L189) and the second registration puts the contribution [in the `contributors` map](https://github.com/eclipse-theia/theia/blob/41a2f5b8d330944498a8e7715d8a30921a73f5d6/packages/plugin-ext/src/main/browser/debug/plugin-debug-service.ts#L79-L89) so that we end up with completely parallel lists, vz. the output of a log in the `init` method of the `PluginDebugService` that sets a timeout to log `{ debuggers: this.debuggers.map(contrib => contrib.type), contributors: Array.from(this.contributors.keys()) }`:

```javascript
{
  debuggers: [
    'extensionHost',
    'node2',
    'pwa-node',
    'node-terminal',
    'pwa-extensionHost',
    'pwa-chrome',
    'pwa-msedge',
    'node'
  ],
  contributors: [
    'extensionHost',
    'node2',
    'pwa-node',
    'node-terminal',
    'pwa-extensionHost',
    'pwa-chrome',
    'pwa-msedge',
    'node'
  ]
}
```

### Additional Information

- Theia Version: 41a2f5b8d330944498a8e7715d8a30921a73f5d6

**Comment by alvsan09** [comment]
@colin-grant-work, I am not quite able to follow the last part, 
Could you please point me to the logging of the identical entries under the PluginDebugService?
 
The types for the contributors vs debuggers is different although I agree we should try to merge them if possible.

**Comment by colin-grant-work** [comment]
Fixed by #10910.
